pkg/api/admin/platformworkloadidentityroleset_validatestatic.go

```diff
@@ -37,27 +42,25 @@ func (sv platformWorkloadIdentityRoleSetStaticValidator) validate(new *PlatformW
 		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "properties.platformWorkloadIdentityRoles", "Must be provided and must be non-empty")
 	}
 
-	errs := []error{}
-
 	for i, r := range new.Properties.PlatformWorkloadIdentityRoles {
 		if r.OperatorName == "" {
-			errs = append(errs, api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, fmt.Sprintf("properties.platformWorkloadIdentityRoles[%d].operatorName", i), "Must be provided"))
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, fmt.Sprintf("properties.platformWorkloadIdentityRoles[%d].operatorName", i), "Must be provided")
```

Comment:
See the changes I made and the commit message for this commit and let me know what you think: https://github.com/Azure/ARO-RP/pull/3608/commits/214350273529cbe16efd0553b2ed36eba7623cc8

Reply:
I see. I think it's fine to fail fast versus having to adjust the error-catching logic. Maybe we could put in a TODO if you think it's worth the cleanup later, but up to you.

Reply:
I'd be cool with cleaning this up now except that I'm not sure how. Do you have any ideas in mind?

Reply:
How about wrapping the joined errors like this?

```golang
return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, errors.Join(errs...).Error())
```

In the first place, we should use `errors.Is` to compare errors instead of type assertion.
If we use it, we can compare wrapped errors.

https://pkg.go.dev/errors#Join

Reply:
I made a ticket of that https://issues.redhat.com/browse/ARO-7922

Reply:
I pushed a commit to wrap multiple validation issues with the `PlatformWorkloadIdentityRoles` together. `api.NewCloudError` works a little bit differently from in your example, so I had to tweak it a bit. Let me know what you all think.